runtime-watcher/tests/e2e/utils/logassert.go

```diff
@@ -0,0 +1,133 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	apicorev1 "k8s.io/api/core/v1"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type LogAsserter struct {
+	controlPlaneConfig *rest.Config
+	runtimeConfig      *rest.Config
+	controlPlaneClient client.Client
+	runtimeClient      client.Client
+}
+
+func NewLogAsserter(
+	controlPlaneConfig,
+	runtimeConfig *rest.Config,
+	controlPlaneClient,
+	runtimeClient client.Client,
+) *LogAsserter {
+	return &LogAsserter{
+		controlPlaneConfig: controlPlaneConfig,
+		runtimeConfig:      runtimeConfig,
+		controlPlaneClient: controlPlaneClient,
+		runtimeClient:      runtimeClient,
+	}
+}
+
+const (
+	controlPlaneNamespace = "kcp-system"
+	remoteNamespace       = "kyma-system"
+	klmPodPrefix          = "klm-controller-manager"
+	klmPodContainer       = "manager"
+	watcherPodPrefix      = "skr-webhook"
+	watcherPodContainer   = "server"
+)
+
+var (
+	errPodNotFound         = errors.New("could not find pod")
+	ErrKLMLogMsgNotFound   = errors.New("log msg was not found in KLM log")
+	ErrWatcherLogsNotFound = errors.New("watcher log was not found since")
+)
+
+func (l *LogAsserter) CheckKLMLogs(ctx context.Context, msg string, since *apimetav1.Time) error {
```

Comment:
```suggestion
func (l *LogAsserter) ContainsMessage(ctx context.Context, msg string, since *apimetav1.Time) error {
```

In this case on the consumer's side it twill be: `logAssert.ContainsMessage(...)`

Reply:
True, KLM needs to stay though. Renamed.